android-kit-base/src/test/kotlin/com/mparticle/kits/KitManagerImplTest.kt

```diff
@@ -602,6 +602,127 @@ class KitManagerImplTest {
+        val filteredKitOptions = MParticleOptions.builder(MockContext())
+            .disableKits(listOf(MParticle.ServiceProviders.ADJUST))
+            .disableKits(listOf(MParticle.ServiceProviders.APPBOY))
+            .disableKits(listOf(MParticle.ServiceProviders.CLEVERTAP))
+            .build()
+
+        val filteredKitIntegrationFactory = KitIntegrationFactory(filteredKitOptions)
+        val kitIntegrationFactory = KitIntegrationFactory(options)
+        val knownIntegrationsField = KitIntegrationFactory::class.java.getDeclaredField("knownIntegrations")
+        knownIntegrationsField.isAccessible = true
+
+        val withoutFilterIntegration = knownIntegrationsField.get(kitIntegrationFactory) as Map<*, *>
+        val filteredKitIntegrations = knownIntegrationsField.get(filteredKitIntegrationFactory) as Map<*, *>
+
+        val knownKitsSize = withoutFilterIntegration.size
+        val filteredKnownKitsSize = filteredKitIntegrations.size
+        Assert.assertEquals(knownKitsSize - 3, filteredKnownKitsSize)
+        // list of All the kits without Filter
+        Assert.assertNotNull(withoutFilterIntegration[MParticle.ServiceProviders.ADJUST])
+        Assert.assertNotNull(withoutFilterIntegration[MParticle.ServiceProviders.APPBOY])
+        Assert.assertNotNull(withoutFilterIntegration[MParticle.ServiceProviders.CLEVERTAP])
+
+        // Filtered kits; the specified kit should not be present (should be null)
+        Assert.assertNull(filteredKitIntegrations[MParticle.ServiceProviders.ADJUST])
+        Assert.assertNull(filteredKitIntegrations[MParticle.ServiceProviders.APPBOY])
+        Assert.assertNull(filteredKitIntegrations[MParticle.ServiceProviders.CLEVERTAP])
+    }
+
+    @Test
+    fun shouldFilterKitsFromKnownIntegrations_When_filter_Is_Empty() {
+        val options = MParticleOptions.builder(MockContext()).build()
+        val filteredKitOptions = MParticleOptions.builder(MockContext())
+            .disableKits(emptyList())
+            .build()
+
+        val filteredKitIntegrationFactory = KitIntegrationFactory(filteredKitOptions)
+        val kitIntegrationFactory = KitIntegrationFactory(options)
+        val knownIntegrationsField = KitIntegrationFactory::class.java.getDeclaredField("knownIntegrations")
+        knownIntegrationsField.isAccessible = true
+
+        val knownIntegrations = knownIntegrationsField.get(kitIntegrationFactory) as Map<*, *>
+        val filteredKnownIntegrations = knownIntegrationsField.get(filteredKitIntegrationFactory) as Map<*, *>
+
+        val knownKitsSize = knownIntegrations.size
+        val filteredKnownKitsSize = filteredKnownIntegrations.size
+        Assert.assertEquals(knownKitsSize, filteredKnownKitsSize)
+    }
+
+    @Test
+    fun shouldIgnoreUnknownKitInFilter() {
+        val options = MParticleOptions.builder(MockContext()).build()
+        val filteredKitOptions = MParticleOptions.builder(MockContext())
+            .disableKits(listOf(1231, 132132))
```

Comment:
i see this API is used here, which makes more sense to me than the one above, so it seems like both APIs work for 
```
        val filteredKitOptions = MParticleOptions.builder(MockContext())
            .filteredKits(listOf(MParticle.ServiceProviders.ADJUST))
            .filteredKits(listOf(MParticle.ServiceProviders.APPBOY))
            .filteredKits(listOf(MParticle.ServiceProviders.CLEVERTAP))
```

and

```
.disableKits(listOf(
     MParticle.ServiceProviders.ADJUST, 
     MParticle.ServiceProviders.APPBOY,
     MParticle.ServiceProviders.CLEVERTAP
     )
)
```

is that correct?